### ISSUE TITLE
Main doesn't handle aliased types

### ISSUE THREAD

**Issue by jastice** [elm-lang/elm-compiler#883]
Elm will not recognize aliased types as valid for use in the `main` function. For example:

```Elm
import Text (asText)
import Graphics.Element (Element)

type alias Fool = Element

thing: Fool
thing = asText "boo"

main = thing
```

will result in a type error:
```
Type Error: 'main' must have one of the following types:
 
    Element, Html, Signal Element, Signal Html
 
Instead 'main' has type:

    Main.Fool
```

using elm-reactor 0.3

**Comment by laszlopandy** [comment]
`String` is not a valid type for main. But the bug still exists for Element. Here is a better example:
```elm
import Text (asText)
import Graphics.Element (Element)

type alias Fool = Element

thing: Fool
thing = asText "boo"

main = thing
```

**Comment by jastice** [comment]
Yep, sorry. Paste error, fixed with your version ;)

**Comment by evancz** [comment]
Should be fixed with 94b3091683b613b9425e5bfb2ec485309f128f7f so that'll be out in the next release!
